fdbclient/BackupContainerFileSystem.actor.cpp

```diff
@@ -613,13 +688,25 @@ class BackupContainerFileSystemImpl {
 					s.restorable = false;
 				if (!desc.contiguousLogEnd.present() || desc.contiguousLogEnd.get() <= s.endVersion)
 					s.restorable = false;
+				// If there is logs gap after contiguousLogEnd, then check whether the current snapshot
+				// can be restored from the logs available after contiguousLogEnd.
+				if (desc.contiguousLogEnd.present() && desc.contiguousLogEnd.get() < s.beginVersion) {
```

Comment:
Do we want to do `if (desc.contiguousLogEnd.present() && desc.contiguousLogEnd.get() <= s.beginVersion)`?
In case `desc.contiguousLogEnd.get() == s.beginVersion`, since contiguousLogEnd is exclusive and the s.beginVersion is inclusive, the data at version `contiguousLogEnd` is the gap? Please correct me if I am wrong. Thanks!

Reply:
This case desc.contiguousLogEnd.get() == s.beginVersion is taken care by line 689. So I eliminated to call unnecessary calling of hasContinuousLogsForSnapshot.

Reply:
My bad, wrongly assumed this is the case of s.beginVersion == s.endVersion.
I will revert this change. Thanks!